libraries/redcore/api/hal/hal.php

```diff
@@ -1220,6 +1244,11 @@ public function processPostData($data, $configuration)
 				$fieldAttributes['defaultValue'] = !empty($fieldAttributes['defaultValue']) ? $fieldAttributes['defaultValue'] : '';
 				$data[$fieldAttributes['name']] = !empty($data[$fieldAttributes['name']]) ? $data[$fieldAttributes['name']] : $fieldAttributes['defaultValue'];
 				$data[$fieldAttributes['name']] = $this->transformField($fieldAttributes['transform'], $data[$fieldAttributes['name']], false);
+
+				if ($removeNulls && $data[$fieldAttributes['name']] == '')
```

Comment:
this should be handled differently. If $removeNulls and "" then just do not transform. Not unset. JTable will take care of the rest to fill in the nulls.

Also I was thinking of using additional attribute isNull="true" in the field attributes that will take care of this, what do you think about using that approach?